### ISSUE TITLE
Notifications and PushNotifications do not work :(

### ISSUE THREAD

**Issue by Bogdan0** [crazyhitty/firebase-chat#6]
I saw there is a method " onPushNotificationEvent" but is never called, same for normal notifications.

Is this feature (notifications) not implemented yet?

Btw, awesome project, I like how you structured it :D and it worsk very good, just the issue with the notifications.

EDIT:
I get the next error  from the notification builder when I send a message : 

E/FcmNotificationBuilder: onResponse: {"multicast_id":8285779204958253584,"success":0,"failure":1,"canonical_ids":0,"results":[{"error":"MissingRegistration"}]}

MissingRegistration.... what could it mean?
EDIT2: 
The first edit was my fault, passed an empty string token as parameter when I start ChatActivity.

Now I pass the correct string token of the user I want to recive the notification. This token, transformed to String looks like  com.google.android.gms.tasks.zzh@4221a2f8 so I pass as parameter something like this.

But I get again an error: 

E/FcmNotificationBuilder: onResponse: {"multicast_id":5901028783144837003,"success":0,"failure":1,"canonical_ids":0,"results":[{"error":"InvalidRegistration"}]}

I have no idea why...

EDIT3: 
I have tryed to store the tokens in my personal database, but it looks like this token change every time I call  getToken(true/false).
Anyway, it didn't make work the notifications.
 
Thank you for your time.

EDIT4:
I  finaly get how works the registration tokens. Actualy I made work the push notifications with FCM

**Comment by crazyhitty** [comment]
Hi Bogdan, were you able to resolve this issue?

Regarding your first question, onPushNotificationEvent method is called via EventBus from MyFirebaseMessagingService.

**Comment by Bogdan0** [comment]
Hi there! Thanks so much for taking your time reading my questions, I have solved all my problems, and everything works fine!

After, I had some problems, cause I didn't really understand how querys of firebase works, actually I have a clear idea of how all this works, and I adapted all to my project. 

I have a big amount of users, and geting datasnapshots from the / of the firebase database is not a good idea anymore due to hight amount of data wasted, but I have improved it with a mysql table in my abitual hosting.

Thank you so much for everything :D
